src/shared/protocol.ts

```diff
@@ -64,6 +64,20 @@ export type RequestOptions = {
    * If not specified, `DEFAULT_REQUEST_TIMEOUT_MSEC` will be used as the timeout.
    */
   timeout?: number;
+
+  /**
+   * If true, receiving a progress notification will reset the request timeout.
+   * This is useful for long-running operations that send periodic progress updates.
+   * Default: false
+   */
+  resetTimeoutOnProgress?: boolean;
+
+  /**
+   * Maximum total time (in milliseconds) to wait for a response, even if progress notifications are received.
+   * Only used when resetTimeoutOnProgress is true.
```

Comment:
It might be a bit confusing that this property only has an effect when another is set. Any reason not to always apply it?